build_tools/packaging/linux/build_package.py

```diff
@@ -0,0 +1,645 @@
+    dest_dir = f"{package_dir}/{config.install_prefix}"
+    pkg_info = get_package_info(pkg_name)
+    generate_control_file(pkg_info, package_dir, config)
+    # check the package is group of basic package or not
+    pkg_list = pkg_info.get("Includes")
+
+    if pkg_list is None:
+        pkg_list = [pkg_info.get("Package")]
+
+    sourcedir_list = []
+    for pkg in pkg_list:
+        dir_list = filter_components_fromartifactory(pkg, config.gfx_arch)
+        sourcedir_list.extend(dir_list)
+
+    for source_path in sourcedir_list:
+        print(source_path)
+        copy_package_contents(source_path, dest_dir)
+
+    pkg_name = update_package_name(pkg_name, config)
+    pkg_name = pkg_name.replace("-devel", "-dev")
+    arch = pkg_info.get("Architecture")
+    version_str = version_to_str(config.rocm_version)
+    debpkg_name = (
+        f"{pkg_name}_{config.rocm_version}.{version_str}-{config.version_suffix}_{arch}"
+    )
+    if config.enable_rpath:
+        print("ENABLE RPATH")
+        subprocess.run(["python3", "runpath_to_rpath.py", package_dir])
+
+    package_with_dpkg_deb(package_dir, config.pkg_dir, debpkg_name)
+
+
+def generate_control_file(pkg_info, pkg_dir, config: PackageConfig):
+    """Function will generate control file for debian package
+
+    Parameters:
+    pkg_info : Package details from the Json file
+    pkg_dir: Directory where package contents and control file is saved
+    config: Configuration object containing package metadata
+
+    Returns: None
+    """
+
+    print("Generate control file")
+    control_dir = f"{pkg_dir}/DEBIAN"
+    os.makedirs(control_dir, exist_ok=True)
+    controlfile = f"{control_dir}/control"
+
+    pkg_name = update_package_name(pkg_info.get("Package"), config)
+    # Only required for debian developement package
+    pkg_name = pkg_name.replace("-devel", "-dev")
+    arch = pkg_info.get("Architecture")
+    description = pkg_info.get("Description")
+    # version = pkg_info.get("Version")
+    version = config.rocm_version
+    section = pkg_info.get("Section")
+    priority = pkg_info.get("Priority")
+    maintainer = pkg_info.get("Maintainer")
+    homepage = pkg_info.get("Homepage")
+    depends_list = pkg_info.get("DEBDepends", [])
+    depends = convert_to_versiondependency(depends_list, config)
+    # Note: The dev package name update should be done after version dependency
+    # Package.json maintains development package name as devel
+    depends = depends.replace("-devel", "-dev")
+
+    env = Environment(loader=FileSystemLoader("."))
+    template = env.get_template("controlfile_template.j2")
+    # Prepare your context dictionary
+    context = {
+        "arch": arch,
+        "depends": depends,
+        "description": description,
+        "homepage": homepage,
+        "maintainer": maintainer,
+        "pkg_name": pkg_name,
+        "priority": priority,
+        "section": section,
+        "version": version,
+    }
+
+    with open(controlfile, "w") as f:
```

Comment:
Not sure this project uses pylint, but usually you should define the encoding when you write a file and pylint will raise it if you don't.

IE
```python
with open(controlfile, 'w', encoding='utf-8') as f:
```